nemesis/arena_prep.py

```diff
@@ -1068,21 +1100,78 @@ def one_mp_weapon():  # Break Damage Limit, or One MP cost
         memory.main.get_item_slot(37) > 200
         or memory.main.get_item_count_slot(memory.main.get_item_slot(37)) < 41
     ):
+        arena_npc()
         nemesis.arena_select.arena_menu_select(1)
         nemesis.arena_select.start_fight(area_index=7, monster_index=0)
         bribe_battle()
         nemesis.arena_select.arena_menu_select(4)
         memory.main.full_party_format("initiative")
-        arena_npc()
         print(
             "###Sleeping powder count:",
             memory.main.get_item_count_slot(memory.main.get_item_slot(37)),
         )
     while not one_mp_ready():
         print("Trying to obtain Gambler's Soul and Purifying Salt items")
+        arena_npc()
         nemesis.arena_select.arena_menu_select(4)
+    
+    # Lv.4 key sphere recovery logic
+    logger.debug(f"lv.4 slot: {memory.main.get_item_slot(84)}")
+    try:
+        logger.debug(f"lv.4 slot: {memory.main.get_item_count_slot(memory.main.get_item_slot(84))}")
```

Comment:
This probably shouldn't be wrapped in a try/except block

Reply:
if the slot is 255, the count will fail, or maybe will pull the wrong value from memory. Maybe need to look at this in the next round of edits.